lib/js/src/rpc/messages/RegisterAppInterface.js

```diff
@@ -0,0 +1,381 @@
+        return this;
+    }
+
+    /**
+    * @return {Language}
+    */
+    getHmiDisplayLanguageDesired() {
+        return this.getObject(Language, RegisterAppInterface.KEY_HMI_DISPLAY_LANGUAGE_DESIRED);
+    }
+
+
+    /**
+    * @param {Array<AppHMIType>} appHMIType
+    * @return {RegisterAppInterface}
+    */
+    setAppHmiType(appHMIType) {
+        //TODO make validate type accept arrays 
+        //this.validateType(AppHMIType, appHMIType);
+
+        this.setParameter(RegisterAppInterface.KEY_APP_HMI_TYPE, appHMIType);
+        return this;
+    }
+
+    /**
+    * @return {Array<AppHMIType>}
+    */
+    getAppHmiType() {
+        return this.getObject(AppHMIType, RegisterAppInterface.KEY_APP_HMI_TYPE);
+    }
+
+    /**
+    * @param {String} hashID the hash ID
+    * @return {RegisterAppInterface}
+    */
+    setHashID(hashID) {
+        //TODO do we need this? this.validateType(String, hashID);
+
+        this.setParameter(RegisterAppInterface.KEY_HASH_ID, hashID);
+        return this;
+    }
+
+    /**
+    * @return {String} the hash ID
+    */
+    getHashID() {
+        return this.getParameter(RegisterAppInterface.KEY_HASH_ID);
+    }
+
+    /**
+     * @param {DeviceInfo} deviceInfo
+     * @return {RegisterAppInterface}
+     */
+    setDeviceInfo(deviceInfo) {
+        this.validateType(DeviceInfo, deviceInfo);
+
+        this.setParameter(RegisterAppInterface.KEY_DEVICE_INFO, deviceInfo);
+        return this;
+    }
+
+    /**
+    * @return {DeviceInfo}
+    */
+    getDeviceInfo() {
+        return this.getParameter(RegisterAppInterface.KEY_DEVICE_INFO);
+    }
+
+
+    /**
+    * @param {String} appName This method should not be accessed directly by developers. Only set the full ID and this param will be set.
+    * @return {RegisterAppInterface}
+    */
+    _setAppId(appId) {
+        //TODO do we need this? this.validateType(String, appId);
+
+        this.setParameter(RegisterAppInterface.KEY_APP_ID, appId);
+        return this;
+    }
+
+    /**
+    * @return {String} the app id
+    */
+    getAppId() {
+        return this.getParameter(RegisterAppInterface.KEY_APP_ID);
+    }
+
+    /**
+    * @param {String} fullAppId 
+    * @return {RegisterAppInterface}
+    */
+    setFullAppId(fullAppId) {
+        //TODO do we need this? this.validateType(String, fullAppId);
+
+        if (fullAppID != null) {
+            fullAppID = fullAppID.toLowerCase();
+            setParameters(RegisterAppInterface.KEY_FULL_APP_ID, fullAppID);
+            let appID;
+            if (fullAppID.length() <= RegisterAppInterface.APP_ID_MAX_LENGTH) {
+                appID = fullAppID;
+            } else {
+                appID = fullAppID.replace("-", "").substring(0, RegisterAppInterface.APP_ID_MAX_LENGTH);
+            }
+            this._setAppId(appID);
+        } else {
+            setParameters(RegisterAppInterface.KEY_FULL_APP_ID, null);
+        }
```

Comment:
Incoming parameter is string, so I believe `validateType` no needed, right? 
There are a little bit advanced logic, in case if we generating such JS code using as input only `MOBILE_API.xml` how can we evaluate where should be added similar logic? Perhaps we should exclude such logic to some external methods like `this.validateType(`?

Reply:
Validating string types was asked previously, per review it wouldn't hurt as we could avoid developers passing in Numbers, Booleans, etc.
These types of cases will come from using the other libraries as examples. While the RPC spec can be used to generate base implementations, there are likely corner cases that will have to be added to RPCs to properly handle cases that were created from spec changes / incorrect OEM implementations / etc. There is nothing the generator can do besides observe the existing code.